StreamChatClient_v3/Controllers/Controller_Tests.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient_v3
+import XCTest
+
+class Controller_Tests: XCTestCase {
+
+    func test_delegateMethodIsCalled() {
+
+        let controller = Controller()
+        let delegate = TestDelegate()
+        controller.stateDelegate = delegate
+
+        XCTAssertEqual(controller.state, .inactive)
+        controller.state = .localDataFetched
+        XCTAssertEqual(controller.state, .localDataFetched)
+    }
```

Comment:
We also should verify the delegate is called on the correct queue. Check how it's done with `QueueAwareDelegate`